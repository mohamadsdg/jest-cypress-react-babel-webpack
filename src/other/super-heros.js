const superHeros = [
  {name: 'Dynaguy', powers: ['disintegrationray', 'fly']},
  {name: 'Apogee', powers: ['gravity control', 'fly']},
  {name: 'Blazestone', powers: ['control offire', 'pyrotechnic discharges']},
  {name: 'Froozone', powers: ['freezewater']},
  {name: 'Mr. Incredible', powers: ['physicalstrength']},
  {name: 'Elastigirl', powers: ['physicalstretch']},
  {name: 'Violet', powers: ['invisibility', 'force fields']},
  {name: 'Dash', powers: ['speed']},
  //   {name: 'Jack-Jack', powers: ['shapeshifting', 'fly']},
]
function getFlyingSuperHeros() {
  return superHeros.filter(hero => {
    return hero.powers.includes('fly')
  })
}
export {getFlyingSuperHeros}
