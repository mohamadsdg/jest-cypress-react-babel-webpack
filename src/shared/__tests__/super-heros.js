// import React from 'react'
import {getFlyingSuperHeros} from '../../other/super-heros'

describe('check filter render', () => {
  // test('static', () => {
  //   const flyingHero = getFlyingSuperHeros()
  //   // console.log(flyingHero)
  //   expect(flyingHero).toEqual([
  //     {name: 'Dynaguy', powers: ['disintegrationray', 'fly']},
  //     {name: 'Apogee', powers: ['gravity control', 'fly']},
  //     // {name: 'Jack-Jack', powers: ['shapeshifting', 'fly']},
  //   ])
  // })

  test('should render flyingHero', () => {
    const flyingHero = getFlyingSuperHeros()
    // console.log(flyingHero)
    expect(flyingHero).toMatchSnapshot()
  })

  test('should render flyingHero inline', () => {
    const flyingHero = getFlyingSuperHeros()
    // console.log(flyingHero)
    expect(flyingHero).toMatchInlineSnapshot(`
      Array [
        Object {
          "name": "Dynaguy",
          "powers": Array [
            "disintegrationray",
            "fly",
          ],
        },
        Object {
          "name": "Apogee",
          "powers": Array [
            "gravity control",
            "fly",
          ],
        },
      ]
    `)
  })
})
