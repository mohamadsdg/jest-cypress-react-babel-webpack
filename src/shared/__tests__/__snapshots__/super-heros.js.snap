// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check filter render should render flyingHero 1`] = `
Array [
  Object {
    "name": "Dynaguy",
    "powers": Array [
      "disintegrationray",
      "fly",
    ],
  },
  Object {
    "name": "Apogee",
    "powers": Array [
      "gravity control",
      "fly",
    ],
  },
]
`;
